pkg/schemas/security_integration.go

```diff
@@ -1,63 +1,38 @@
-	"oauth_use_secondary_roles":             DescribePropertyListSchema,
-	"pre_authorized_roles_list":             DescribePropertyListSchema,
-	"blocked_roles_list":                    DescribePropertyListSchema,
-	"oauth_issue_refresh_tokens":            DescribePropertyListSchema,
-	"oauth_refresh_token_validity":          DescribePropertyListSchema,
-	"network_policy":                        DescribePropertyListSchema,
-	"oauth_client_rsa_public_key_fp":        DescribePropertyListSchema,
-	"oauth_client_rsa_public_key_2_fp":      DescribePropertyListSchema,
-	"comment":                               DescribePropertyListSchema,
-	"oauth_client_id":                       DescribePropertyListSchema,
-	"oauth_authorization_endpoint":          DescribePropertyListSchema,
-	"oauth_token_endpoint":                  DescribePropertyListSchema,
-	"oauth_allowed_authorization_endpoints": DescribePropertyListSchema,
-	"oauth_allowed_token_endpoints":         DescribePropertyListSchema,
-}
+var (
+	SecurityIntegrationDescribeSchema = helpers.MergeMaps(
+		DescribeApiAuthSecurityIntegrationSchema,
+		DescribeExternalOauthSecurityIntegrationSchema,
+		DescribeOauthIntegrationForCustomClients,
+		DescribeOauthIntegrationForPartnerApplications,
+		DescribeSaml2IntegrationSchema,
+		DescribeScimSecurityIntegrationSchema,
+	)
+	allSecurityIntegrationPropertiesNames = helpers.ConcatSlices(
+		ApiAuthenticationPropertiesNames,
+		ExternalOauthPropertiesNames,
+		OauthIntegrationForCustomClientsPropertiesNames,
+		OauthIntegrationForPartnerApplicationsPropertiesNames,
+		Saml2PropertiesNames,
+		ScimPropertiesNames,
+	)
+)
 
 func SecurityIntegrationsDescriptionsToSchema(integrationProperties []sdk.SecurityIntegrationProperty) map[string]any {
 	securityIntegrationProperties := make(map[string]any)
-	for _, property := range integrationProperties {
-		property := property
-		switch property.Name {
-		case "OAUTH_CLIENT_TYPE",
-			"OAUTH_REDIRECT_URI",
-			"ENABLED",
-			"OAUTH_ALLOW_NON_TLS_REDIRECT_URI",
-			"OAUTH_ENFORCE_PKCE",
-			"OAUTH_USE_SECONDARY_ROLES",
-			"PRE_AUTHORIZED_ROLES_LIST",
-			"BLOCKED_ROLES_LIST",
-			"OAUTH_ISSUE_REFRESH_TOKENS",
-			"OAUTH_REFRESH_TOKEN_VALIDITY",
-			"NETWORK_POLICY",
-			"OAUTH_CLIENT_RSA_PUBLIC_KEY_FP",
-			"OAUTH_CLIENT_RSA_PUBLIC_KEY_2_FP",
-			"COMMENT",
-			"OAUTH_CLIENT_ID",
-			"OAUTH_AUTHORIZATION_ENDPOINT",
-			"OAUTH_TOKEN_ENDPOINT",
-			"OAUTH_ALLOWED_AUTHORIZATION_ENDPOINTS",
-			"OAUTH_ALLOWED_TOKEN_ENDPOINTS":
-			securityIntegrationProperties[strings.ToLower(property.Name)] = []map[string]any{SecurityIntegrationPropertyToSchema(&property)}
-		default:
-			log.Printf("[WARN] unexpected property %v returned from Snowflake", property.Name)
+	for _, desc := range integrationProperties {
+		desc := desc
+		if slices.Contains(allSecurityIntegrationPropertiesNames, desc.Name) {
+			securityIntegrationProperties[strings.ToLower(desc.Name)] = []map[string]any{SecurityIntegrationPropertyToSchema(&desc)}
 		}
```

Comment:
no else ?

Reply:
Yup, let's leave this default with WARN pls

Reply:
Added.